tests/generation/test_contrastive_search.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+
+import torch.nn.functional as F
+from parameterized import parameterized
+
+from transformers import (
+    GPT2LMHeadModel,
+    GPT2Tokenizer,
+)
+from transformers.testing_utils import (
+    require_torch,
+    slow,
+    torch_device,
+)
+
+
+@require_torch
+class ContrastiveSearchTest(unittest.TestCase):
+    @parameterized.expand(
+        [
+            (False, 4, 0.6),
+            (True, 6, 0.4),
+        ]
+    )
+    @slow
+    def test_batch_contrastive_search_gpt2(self, low_memory, top_k, penalty_alpha):
```

Comment:
Please move this test inside `GenerationIntegrationTests`, in `tests/generation/test_utils.py` (we don't separate tests according to the decoding strategy)